pmpsui/pmps.py

```diff
@@ -94,12 +103,46 @@ def __init__(self, parent=None, args=None, macros=None):
         self._channels = []
         self.setup_ui()
 
+        self.splash.finish(self)
+        self.splash = None
+
+    def setup_splash(self) -> None:
+        pixmap = QPixmap(str(Path(__file__).parent.parent / 'pmps_splash.png'))
+        self.splash = PMPSSplashScreen(pixmap)
+        self.splash.set_message_rect(QtCore.QRect(250, 100, 300, 150),
+                                     alignment=QtCore.Qt.AlignLeft)
+
+        splash_font = QtGui.QFont("Arial", 12, QtGui.QFont.Bold)
+        self.splash.setFont(splash_font)
+        self.splash.show_message('PMPS UI setup beginning...')
+        self.splash.set_progress(0)
+        self.splash.show()
+
+    def update_splash_message(
+        self,
+        msg: str,
+        progress: Optional[int]=None
+    ) -> None:
+        if self.splash is None:
+            return
+
+        if progress is not None:
+            self.splash.set_progress(progress)
+        self.splash.show_message(msg)
```

Comment:
Interestingly, switching the order did make the splash progress consistent with the text